src/v/cloud_storage/segment_meta_cstore.h

```diff
@@ -827,16 +829,18 @@ class deltafor_column_impl
 /// specialization is for xor-delta algorithm. It doesn't allow skipping
 /// frames so all search operations require full scan. Random access by
 /// index can skip frames since indexes are monotonic.
-template<class value_t>
-class deltafor_column<value_t, details::delta_xor>
+template<class value_t, size_t max_frame_size>
+class deltafor_column<value_t, details::delta_xor, max_frame_size>
   : public deltafor_column_impl<
       value_t,
       details::delta_xor{},
-      deltafor_column<value_t, details::delta_xor>> {
+      deltafor_column<value_t, details::delta_xor, max_frame_size>,
+      max_frame_size> {
     using base_t = deltafor_column_impl<
       value_t,
       details::delta_xor{},
-      deltafor_column<value_t, details::delta_xor>>;
+      deltafor_column<value_t, details::delta_xor, max_frame_size>,
+      max_frame_size>;
```

Comment:
i wonder if this duplication of max_frame_size points at a slightly different decomposition of the types being possible?

for example, would it make logical sense for the deltafor_column to expose its configured max size as a constexpr field and then the column_impl would use that?

i'm assuming that it would be problematic for the two max sizes to be different?

Reply:
I don't think it's problematic because the value is pushed from top to bottom during instantiation. The `deltafor_column` instantiates `deltafor_column_impl` which actually uses the value. The stuff in the `column_store` doesn't really care about the value.

Reply:
Since the actual impl of `append()` is in the `_impl` class, I don't think making a constexpr field will change much, since it still needs to get passed to the `_impl`. I dont feel strongly about it at all, but perhaps an alternative is something like:

- deltafor column: owns the `_frames`, is also an appender and a seeker, and instantiates them to point at `_frames`
- frame appender: references the frames, appends to the column, templatized on max_frame_size, uses it to roll new frames, identical impl regardless of algo
- frame seeker: references the frames, templatized on algo, seeks based on which algo is templatized

Though I'm not convinced further decomposition adds much beyond maybe it clarifies the roll of the frame size, and solidifies the difference the algos' different seek implementations.